### DISCUSSION TITLE
Question: should KAS Grants to Attribute Definitions or to Values be singular (1 : 1) or multiple?

### DISCUSSION THREAD

**Discussion by jakedoublev** [opentdf/platform#1297] [Platform Q&A]
The current policy protos and db allow for multiple grants of a KAS to a single [Attribute Definition](https://github.com/opentdf/platform/blob/main/service/policy/objects.proto#L40) or to a [single Value](https://github.com/opentdf/platform/blob/main/service/policy/objects.proto#L72). The db logic also allows this behavior.

I am wondering if this is intended or if we should only support a singular grant to each individual part of an attribute?

**Comment by jakedoublev** [comment]
Curious to hear your thoughts @dmihalcik-virtru, @strantalis, @jp-ayyappan.

**Reply by dmihalcik-virtru** [reply]
We support multiple grants to allow a single attribute or value to indicate a shared key, e.g. to allow two organizations to independently control access to a document. This could indicate a shared project or topic of interest.

**Reply by jrschumacher** [reply]
Does this denote an AND or OR split?

**Comment by jp-ayyappan** [comment]
The need is to be able to drive encryption keys used based on attribute values in TDF. I have tried to map out the full set of applications below.


### Case 1 - 1:1 mapping
Where an attribute value has one key mapped to it
![image](https://github.com/user-attachments/assets/dbd3b9c9-1d0b-4b2e-bd86-01676fdba421)
### Case 1 - 1:many OR mapping
Where an attribute value has more than one key mapped to it; but only one key is needed to decrypt
![image](https://github.com/user-attachments/assets/33ecd1d1-5166-4508-a9e6-1b483eef59a0)

### Case 1 - 1:many AND mapping
Where an attribute value has more than one key mapped to it; and all the keys are needed to decrypt
![image](https://github.com/user-attachments/assets/eb5cd351-663a-48e3-8d28-bc7e2eddd3b2)
